perceval/utils/density_matrix.py

```diff
@@ -363,6 +364,48 @@ def sample(self, count: int = 1):
         output = random.choices(self.reverse_index, list(self.mat.diagonal()), k=count)
         return output
 
+    @staticmethod
+    def _get_annihilated_fockstate(fockstate, m, n_photon):
+        """
+        give the fockstate after loss of n_photon in the mode m
+        """
+
+        listed_fs = list(fockstate)
+        if listed_fs[m] <= n_photon:
+            listed_fs[m] = 0
+        else:
+            listed_fs[m] -= n_photon
+        return BasicState(listed_fs)
+
+    def _construct_loss_operators(self, mode: int, p:float):
```

Comment:
I am a little confused with how the loss is modeled here; the parameter `p` here comes from `probs` - does it mean that the user inputs some sort of loss probability?

Reply:
yes
I agree that prob might not be the best name.